logger/factory_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logger
+
+import (
+	"github.com/stretchr/testify/suite"
+	"github.com/uber/ringpop-go/test/mocks"
+	"testing"
+)
+
+type LoggerFactorySuite struct {
```

Comment:
I don't really understand what you mean. Why are they brittle?

In the example you give, if you change the caching strategy, from what I can tell in the tests below, nothing will break, because you haven't changed the contract of the `LoggerFactory.Logger` method.

Any change to field names will most certainly break, though. Fortunately it's just a find-replace fix which is usually quick. However, where possible, we should avoid using internals (like `min` below) and instead test through public APIs.